app/src/main/java/com/github/epfl/meili/photo/CameraActivity.kt

```diff
@@ -0,0 +1,251 @@
+                        isPressed = false
+                    }, PRESS_DELAY)
+                }
+                true
+            }
+            else -> super.onKeyDown(keyCode, event)
+        }
+    }
+
+    /**
+     * Creates the touch listener for the previewView
+     */
+    private fun getPreviewTouchListener(): (View, MotionEvent) -> Boolean {
+        // Pinch to zoom
+        val pinchListener = object : ScaleGestureDetector.SimpleOnScaleGestureListener() {
+            override fun onScale(detector: ScaleGestureDetector): Boolean {
+                val currentZoomRatio = camera.cameraInfo.zoomState.value?.zoomRatio ?: 0F
+                camera.cameraControl.setZoomRatio(currentZoomRatio * detector.scaleFactor)
+                return true
+            }
+        }
+        val scaleGestureDetector = ScaleGestureDetector(applicationContext, pinchListener)
+
+        return { view: View, motionEvent: MotionEvent ->
+            view.performClick()
+            // Touch to focus camera
+            scaleGestureDetector.onTouchEvent(motionEvent)
+            when (motionEvent.action) {
+                MotionEvent.ACTION_DOWN -> true
+                MotionEvent.ACTION_UP -> {
+                    val factory = previewView.meteringPointFactory
+                    val point = factory.createPoint(motionEvent.x, motionEvent.y)
+                    val action = FocusMeteringAction.Builder(point).build()
+                    camera.cameraControl.startFocusAndMetering(action)
+                    true
+                }
+                else -> false
+            }
+        }
+    }
+
+    /** Sets up camera */
+    private fun startCamera() {
+        val cameraProviderFuture = ProcessCameraProvider.getInstance(this)
+
+        cameraProviderFuture.addListener({
+            cameraProvider = cameraProviderFuture.get() // Guaranteed to exist
+
+            // Setup Preview use case --> to display the preview to the screen
+            preview = Preview.Builder().build()
+
+            // Setup Image Capture use case --> allows user to take photos
+            imageCapture = ImageCapture.Builder()
+                .setCaptureMode(ImageCapture.CAPTURE_MODE_MINIMIZE_LATENCY)
+                .build()
+
+            // Setup which camera to select (default is back)
+            cameraSelector = CameraSelector.Builder()
+                .requireLensFacing(CameraSelector.LENS_FACING_BACK)
+                .build()
+
+            // Display preview in activity
+            val previewView = findViewById<PreviewView>(R.id.camera_preview)
+            preview.setSurfaceProvider(previewView.surfaceProvider)
+
+            // Attach the lifecycle of the camera to the activity's lifecycle
+            camera = cameraProvider.bindToLifecycle(
+                this as LifecycleOwner, cameraSelector, preview, imageCapture
+            )
+
+        }, ContextCompat.getMainExecutor(this))
+    }
+
+    /** Takes a picture */
```

Comment:
Just a little nitpick: while comments are nice, obvious ones just clutter the code imo. It would be great if you could this and other such comments in this file!

Reply:
Done